counter/views/click.py

```diff
@@ -413,9 +394,75 @@ def get_context_data(self, **kwargs):
                 )
         kwargs["customer"] = self.customer
         kwargs["basket_total"] = self.sum_basket(self.request)
-        kwargs["refill_form"] = self.refill_form or RefillForm()
-        kwargs["barmens_can_refill"] = self.object.can_refill()
-        kwargs["student_card_fragment"] = StudentCardFormView.get_template_data(
-            self.customer
-        ).render(self.request)
+
+        if self.object.type == "BAR":
+            kwargs["student_card_fragment"] = StudentCardFormView.get_template_data(
+                self.customer
+            ).render(self.request)
+
+        if self.object.can_refill():
+            kwargs["refilling_fragment"] = RefillingCreateView.get_template_data(
+                self.customer
+            ).render(self.request)
         return kwargs
+
+
+class RefillingCreateView(FormView):
+    """This is a fragment only view which integrates with counter_click.jinja"""
+
+    form_class = RefillForm
+    template_name = "counter/fragments/create_refill.jinja"
+
+    @classmethod
+    def get_template_data(
+        cls, customer: Customer, *, form_instance: form_class | None = None
+    ) -> FormFragmentTemplateData[form_class]:
+        return FormFragmentTemplateData[cls.form_class](
+            form=form_instance if form_instance else cls.form_class(),
+            template=cls.template_name,
+            context={
+                "action": reverse_lazy(
+                    "counter:refilling_create", kwargs={"customer_id": customer.pk}
+                ),
+            },
+        )
```

Comment:
```suggestion
        return FormFragmentTemplateData(
            form=form_instance if form_instance else cls.form_class(),
            template=cls.template_name,
            context={
                "action": reverse_lazy(
                    "counter:refilling_create", kwargs={"customer_id": customer.pk}
                ),
            },
        )
```
Préciser le `[cls.form_class]` me parait un peu douteux. Je pense que c'est pas faux, mais c'est quand même une source de confusion pour l'analyse statique. 

Pour le typage en lui-même, on a déjà précisé le type générique de la structure en signature de la fonction, donc pas nécessaire de le remettre ici.